cxxrtl: error: use of undeclared identifier 'debug_item'

## Steps to reproduce the issue

I have a minimal blinky design to test CXXRTL VCD wave dumping:

[blink_vcd.tar.gz](https://github.com/YosysHQ/yosys/files/4750256/blink_vcd.tar.gz)

I run `make`.

## Expected behavior

It compiles.

## Actual behavior

Both g++ and clang++ fail to compile `cxxrtl_vcd_capi.cc`.

clang++: 
```
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc:47:24: error: use of undeclared identifier 'debug_item'
        vcd->writer.add(name, debug_item(*object));
                              ^
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc:58:38: error: unknown type name 'debug_item'; did you mean 'cxxrtl::debug_item'?
                [=](const std::string &name, const debug_item &item) {
                                                   ^~~~~~~~~~
                                                   cxxrtl::debug_item
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:724:8: note: 'cxxrtl::debug_item' declared here
struct debug_item : ::cxxrtl_object {
       ^
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:146:42: warning: shift count >= width of type [-Wshift-count-overflow]
                                : data[shift_chunks + result.chunks] << (chunk::bits - shift_bits);
                                                                     ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:444:14: note: in instantiation of function template specialization 'cxxrtl::value<1>::rtrunc<1>' requested here
                        .template rtrunc<T::bits - Start>()
                                  ^
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:457:43: note: in instantiation of member function 'cxxrtl::slice_expr<const cxxrtl::value<1>, 0, 0>::operator value' requested here
                return static_cast<const value<bits> &>(*this);
                                                        ^
./blink.cpp:28:33: note: in instantiation of member function 'cxxrtl::slice_expr<const cxxrtl::value<1>, 0, 0>::val' requested here
                return !prev_p_clk.slice<0>().val() && p_clk.slice<0>().val();
                                              ^
In file included from main.cpp:6:
In file included from ./blink.cpp:1:
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:146:7: warning: array index 1 is past the end of the array (which contains 1 element) [-Warray-bounds]
                                : data[shift_chunks + result.chunks] << (chunk::bits - shift_bits);
                                  ^    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:77:2: note: array 'data' declared here
        chunk::type data[chunks] = {};
        ^
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:151:34: warning: shift count >= width of type [-Wshift-count-overflow]
                                : data[shift_chunks + n - 1] << (chunk::bits - shift_bits);
                                                             ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:146:7: warning: array index 1 is past the end of the array (which contains 1 element) [-Warray-bounds]
                                : data[shift_chunks + result.chunks] << (chunk::bits - shift_bits);
                                  ^    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:444:14: note: in instantiation of function template specialization 'cxxrtl::value<8>::rtrunc<1>' requested here
                        .template rtrunc<T::bits - Start>()
                                  ^
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:457:43: note: in instantiation of member function 'cxxrtl::slice_expr<cxxrtl::value<8>, 7, 7>::operator value' requested here
                return static_cast<const value<bits> &>(*this);
                                                        ^
./blink.cpp:58:41: note: in instantiation of member function 'cxxrtl::slice_expr<cxxrtl::value<8>, 7, 7>::val' requested here
        p_led.next = p_counter.curr.slice<7>().val();
                                               ^
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:77:2: note: array 'data' declared here
        chunk::type data[chunks] = {};
        ^
5 warnings and 2 errors generated.

```

g++:
```
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc: In function ‘void cxxrtl_vcd_add(cxxrtl_vcd, const char*, cxxrtl_object*)’:
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc:47:24: error: ‘debug_item’ was not declared in this scope
  vcd->writer.add(name, debug_item(*object));
                        ^~~~~~~~~~
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc:47:24: note: suggested alternatives:
In file included from blink.cpp:1:0,
                 from main.cpp:6:
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:724:8: note:   ‘cxxrtl::debug_item’
 struct debug_item : ::cxxrtl_object {
        ^~~~~~~~~~
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl.h:724:8: note:   ‘cxxrtl::debug_item’
In file included from blink.cpp:9:0,
                 from main.cpp:6:
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc: In function ‘void cxxrtl_vcd_add_from_if(cxxrtl_vcd, cxxrtl_handle, void*, int (*)(void*, const char*, const cxxrtl_object*))’:
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc:58:38: error: ‘debug_item’ does not name a type
   [=](const std::string &name, const debug_item &item) {
                                      ^~~~~~~~~~
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc: In lambda function:
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc:60:5: error: expected ‘{’ before ‘;’ token
   });
     ^
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc: In function ‘void cxxrtl_vcd_add_from_if(cxxrtl_vcd, cxxrtl_handle, void*, int (*)(void*, const char*, const cxxrtl_object*))’:
/home/tom/tools/yosys/backends/cxxrtl/cxxrtl_vcd_capi.cc:60:5: error: expected ‘)’ before ‘;’ token
```

## Environment

* Ubuntu 18.04
* g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
* clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)

## Fix

```c++
diff --git a/backends/cxxrtl/cxxrtl_vcd_capi.cc b/backends/cxxrtl/cxxrtl_vcd_capi.cc
index 46e4f1c4..52a9198b 100644
--- a/backends/cxxrtl/cxxrtl_vcd_capi.cc
+++ b/backends/cxxrtl/cxxrtl_vcd_capi.cc
@@ -44,7 +44,7 @@ void cxxrtl_vcd_add(cxxrtl_vcd vcd, const char *name, cxxrtl_object *object) {
        // Note the copy. We don't know whether `object` came from a design (in which case it is
        // an instance of `debug_item`), or from user code (in which case it is an instance of
        // `cxxrtl_object`), so casting the pointer wouldn't be safe.
-       vcd->writer.add(name, debug_item(*object));
+       vcd->writer.add(name, cxxrtl::debug_item(*object));
 }
 
 void cxxrtl_vcd_add_from(cxxrtl_vcd vcd, cxxrtl_handle handle) {
@@ -55,7 +55,7 @@ void cxxrtl_vcd_add_from_if(cxxrtl_vcd vcd, cxxrtl_handle handle, void *data,
                                                                                                                int (*filter)(void *data, const char *name,
                                                                                                                              const cxxrtl_object *object)) {
        vcd->writer.add(cxxrtl_debug_items_from_handle(handle),
-               [=](const std::string &name, const debug_item &item) {
+               [=](const std::string &name, const cxxrtl::debug_item &item) {
                        return filter(data, name.c_str(), static_cast<const cxxrtl_object*>(&item));
                });
 }
```

--- a/backends/cxxrtl/cxxrtl_vcd_capi.cc
+++ b/backends/cxxrtl/cxxrtl_vcd_capi.cc
@@ -44,7 +44,7 @@ void cxxrtl_vcd_add(cxxrtl_vcd vcd, const char *name, cxxrtl_object *object) {
 	// Note the copy. We don't know whether `object` came from a design (in which case it is
 	// an instance of `debug_item`), or from user code (in which case it is an instance of
 	// `cxxrtl_object`), so casting the pointer wouldn't be safe.
-	vcd->writer.add(name, debug_item(*object));
+	vcd->writer.add(name, cxxrtl::debug_item(*object));
 }
 
 void cxxrtl_vcd_add_from(cxxrtl_vcd vcd, cxxrtl_handle handle) {
@@ -55,7 +55,7 @@ void cxxrtl_vcd_add_from_if(cxxrtl_vcd vcd, cxxrtl_handle handle, void *data,
 														int (*filter)(void *data, const char *name,
 														              const cxxrtl_object *object)) {
 	vcd->writer.add(cxxrtl_debug_items_from_handle(handle),
-		[=](const std::string &name, const debug_item &item) {
+		[=](const std::string &name, const cxxrtl::debug_item &item) {
 			return filter(data, name.c_str(), static_cast<const cxxrtl_object*>(&item));
 		});
 }

